llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -378,6 +378,27 @@ AArch64TargetLowering::AArch64TargetLowering(const TargetMachine &TM,
     addQRTypeForNEON(MVT::v2i64);
     addQRTypeForNEON(MVT::v8f16);
     addQRTypeForNEON(MVT::v8bf16);
+  } else if (Subtarget->hasNEON() || Subtarget->useSVEForFixedLengthVectors()) {
+    addRegisterClass(MVT::v16i8, &AArch64::FPR8RegClass);
+    addRegisterClass(MVT::v8i16, &AArch64::FPR16RegClass);
+
+    addRegisterClass(MVT::v2f32, &AArch64::FPR64RegClass);
```

Comment:
I don't really like the duplication here.  What about passing `isNeonAvailable()` into `addDRTypeForNEON` and `addQRTypeForNEON` to toggle the call to `addTypeForNEON`?

Reply:
I've moved this into (the renamed functions) `addQRType` and `addDRType` now.